cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/AppResourceWatcherThread.java

```diff
@@ -95,13 +105,62 @@ Cancellable addListener(ResourceChangeListener<T> listener) {
   protected Watchable<T> createWatchable(Type resourceType, String namespace,
                                          @Nullable String labelSelector) throws IOException, ApiException {
     ApiClient apiClient = getApiClient();
-    CustomObjectsApi api = new CustomObjectsApi(apiClient);
-    Call call = api.listNamespacedCustomObjectCall(group, version, namespace, plural, null, null, null,
-                                                   labelSelector, null, null, null, true, null);
+    Call call;
+    if (resourceType.equals(V1Job.class)) {
```

Comment:
Why need all these `if`? The `listNamespacedCustomObjectCall` doesn't work??

Reply:
The problem is that `CustomObjectsApi.listNamespacedCustomObject` returns an `Object`, so it's hard to extract a resource version from it.

The other APIs used in the `if` return specific types that allow us to extract the resource version. For example, `BatchV1Api.listNamespacedJob` returns a `V1JobList` - we can get the resource version of this list call from its metadata and pass it to `BatchV1Api.listNamespacedJobCall`.